lib/exsoda/config.ex

```
@@ -0,0 +1,61 @@
+defmodule Exsoda.Config do
+  @moduledoc """
+  This module handles fetching values from the config with some additional niceties
+
+  This config module copied with <3 from: https://gist.github.com/bitwalker/a4f73b33aea43951fe19b242d06da7b9
+  """
+
+  @doc """
+  Fetches a value from the config, or from the environment if {:system, "VAR"}
+  is provided.
+  An optional default value can be provided if desired.
+  ## Example
+      iex> {test_var, expected_value} = System.get_env |> Enum.take(1) |> List.first
+      ...> Application.put_env(:myapp, :test_var, {:system, test_var})
+      ...> ^expected_value = #{__MODULE__}.get(:myapp, :test_var)
+      ...> :ok
+      :ok
+      iex> Application.put_env(:myapp, :test_var2, 1)
+      ...> 1 = #{__MODULE__}.get(:myapp, :test_var2)
+      1
+      iex> :default = #{__MODULE__}.get(:myapp, :missing_var, :default)
+      :default
+  """
+  @spec get(atom, atom, term | nil) :: term
+  def get(app, key, default \\ nil) when is_atom(app) and is_atom(key) do
+    case Application.get_env(app, key) do
+      {:system, env_var} ->
+        case System.get_env(env_var) do
+          nil -> default
+          val -> val
+        end
+      {:system, env_var, preconfigured_default} ->
+        case System.get_env(env_var) do
+          nil -> preconfigured_default
+          val -> val
+        end
+      nil ->
+        default
+      val ->
+        val
+    end
+  end
+
+  @doc """
+  Same as get/3, but returns the result as an integer.
+  If the value cannot be converted to an integer, the
+  default is returned instead.
+  """
```

Do we want it to be a silent failure if the key can't be converted to an int?

---

Added a warning log message